src/figuresClasses.ts

```diff
@@ -1,19 +1,89 @@
-export interface Figure {
+enum Color {
+  'red', 'green', 'blue'
+}
 
+export interface Figure {
+  shape: 'triangle' | 'circle' |'rectangle';
+  color: Color;
+  getArea: Function;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+    public shape: 'triangle',
+  ) {
+    if ((this.a <= 0 || this.b <= 0 || this.c <= 0)
```

Comment:
```suggestion
    if ((a <= 0 || b <= 0 || c <= 0)
```
Do these checks in the constructor this way, as you want check the coming variables anyway

Reply:
done